### DISCUSSION TITLE
Render UserControll inside Window and change it by clicking button

### DISCUSSION THREAD

**Discussion by Orholt** [AvaloniaUI/Avalonia#6413] [Q&A] [answered]
Hi, I'm using **Avalonia** Framework with .net 5.0, I have few UserControls and Main Window. Is there any proper way to change active UserControl visible in main window by pressing button?

Now, I can have only one static UserControl and my MainWindow.xaml look's like this:

```xml
... some XML ...
<local:DeviceListControlView x:Name="ControlViewMain" Grid.Column="1"></local:DeviceListControlView>
... some XML
```

And It look's like this (Attached Image), how can I make it work as this: (Button Click Event) -> (Changing UserControl)

![image](https://user-images.githubusercontent.com/42500203/129386173-a8c88924-ce04-4ba1-a2fa-1ca82e7cb64a.png)

**Comment by jp2masa** [comment, marked as answer]
You can use a `ContentControl`:

```xaml
<ContentControl Grid.Column="1" Content="{Binding CurrentControl}" />
```

Or, if you're using pure MVVM, `CurrentControl` would be a viewmodel and you would have data templates for each viewmodel type, for example in the `ContentControl` itself (could also be on `App` or any parent):

```xaml
<ContentControl Grid.Column="1" Content="{Binding CurrentControl}">
    <ContentControl.DataTemplates>
        <DataTemplate DataType="vm:DeviceListControlViewModel">
            <local:DeviceListControlView />
        </DataTemplate>
    </ContentControl.DataTemplates>
</ContentControl>
```

If you don't want to use bindings, you can just set the `Content` property from code behind.

**Reply by Orholt** [reply]
Hi, thanks for answering my question but could you make example code for me, cause I think i still don't understand how can I do it.

Let me describe my problem better:
I have Avalonia application as shown in question with one main **Window** and like 6 different **UserControlls**.
What I need to do: *After pressing coresponding button on the left side of main Window I need to "replace" curren visible  UserControll*

So in conclusion I want something like this:

![image](https://user-images.githubusercontent.com/42500203/129444951-4b925530-6465-4cff-9bf0-29bc3e9f5431.png)

**Reply by jp2masa** [reply]
[AvaloniaDemo.zip](https://github.com/AvaloniaUI/Avalonia/files/6986415/AvaloniaDemo.zip)

**Reply by Orholt** [reply]
Thank you very much
